In [1]:
import numpy as np
import pandas as pd

In [4]:
df = pd.read_csv('sales.csv')
df.head()

InvoiceNo StockCode                          Description  Quantity  \
0     536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1     536365     71053                  WHITE METAL LANTERN         6   
2     536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3     536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4     536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   

           InvoiceDate  UnitPrice  CustomerID         Country  
0  2010-12-01 08:26:00       2.55     17850.0  United Kingdom  
1  2010-12-01 08:26:00       3.39     17850.0  United Kingdom  
2  2010-12-01 08:26:00       2.75     17850.0  United Kingdom  
3  2010-12-01 08:26:00       3.39     17850.0  United Kingdom  
4  2010-12-01 08:26:00       3.39     17850.0  United Kingdom

In [8]:
df.shape

(375187, 8)

- Number of data points (rows) 375187
- Number of features (columns) 8

In [5]:
#Checking Null/missing values

df.isna().sum()

InvoiceNo      0
StockCode      0
Description    0
Quantity       0
InvoiceDate    0
UnitPrice      0
CustomerID     0
Country        0
dtype: int64

- No missing/null values found

In [15]:
df.duplicated().sum()

5021

In [22]:
df.drop_duplicates(keep='first', inplace=True)

In [23]:
df.duplicated().sum()

0

- Removed all duplicate data points

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 375187 entries, 0 to 375186
Data columns (total 8 columns):
InvoiceNo      375187 non-null int64
StockCode      375187 non-null object
Description    375187 non-null object
Quantity       375187 non-null int64
InvoiceDate    375187 non-null object
UnitPrice      375187 non-null float64
CustomerID     375187 non-null float64
Country        375187 non-null object
dtypes: float64(2), int64(2), object(4)
memory usage: 22.9+ MB


- StockCode, Description, InvoiceDate, Country are object data types
- InvoiceNo and Quantity are int64 data types
- UnitPrice and CustomerID are float64 data types

In [10]:
df.describe()

InvoiceNo       Quantity      UnitPrice     CustomerID
count  375187.000000  375187.000000  375187.000000  375187.000000
mean   560720.262192      13.584978       2.318388   15306.779446
std     13104.443738     184.647865       1.825628    1710.647733
min    536365.000000       1.000000       0.001000   12346.000000
25%    549291.000000       2.000000       1.060000   13975.000000
50%    562031.000000       6.000000       1.650000   15187.000000
75%    572194.000000      12.000000       2.950000   16808.000000
max    581587.000000   80995.000000       8.290000   18287.000000

In [24]:
df.to_csv('sales_final.csv', encoding='utf-8', index=False)

- Finally exported cleaned data to csv file to perform RFM analysis